agent/bls.mjs

```diff
@@ -1623,14 +1693,498 @@ async function do_sign_u256( u256, details, fn ) {
+                    const strErrorMessage =
+                        strLogPrefix + cc.fatal( "CRITICAL ERROR:" ) +
+                        cc.error( " JSON RPC call to SGX failed, RPC call reported error: " ) + cc.warning( owaspUtils.extract_error_message( err ) ) +
+                        cc.error( ", stack is:" ) + "\n" + cc.stack( err_js.stack ) +
+                        "\n";
+                    log.write( strErrorMessage );
+                    details.write( strErrorMessage );
+                    await joCall.disconnect();
+                    throw err_js;
+                }
+                details.write( strLogPrefix + cc.debug( "Call to " ) + cc.info( "SGX" ) + cc.debug( " done, answer is: " ) + cc.j( joOut ) + "\n" );
+                joSignResult = joOut;
+                if( joOut.result != null && joOut.result != undefined && typeof joOut.result == "object" )
+                    joSignResult = joOut.result;
+                if( joOut.signResult != null && joOut.signResult != undefined && typeof joOut.signResult == "object" )
+                    joSignResult = joOut.signResult;
+                if( "errorMessage" in joSignResult &&
+                    typeof joSignResult.errorMessage == "string" &&
+                    joSignResult.errorMessage.length > 0
+                ) {
+                    const strError = "BLS signing finished with error: " + joSignResult.errorMessage;
+                    joRetVal.error = strError;
+                    const strErrorMessage =
+                        strLogPrefix + cc.fatal( "CRITICAL ERROR:" ) +
+                        cc.error( " BLS signing(1) finished with error: " ) + cc.warning( joSignResult.errorMessage ) + "\n";
+                    log.write( strErrorMessage );
+                    details.write( strErrorMessage );
+                    await joCall.disconnect();
+                    throw new Error( strError );
+                }
+                joSignResult.error = null;
+                await joCall.disconnect();
+            } ); // joCall.call ...
+        } ); // rpcCall.create ...
+    } catch ( err ) {
+        const strError = owaspUtils.extract_error_message( err );
+        joSignResult = { };
+        joSignResult.error = strError;
+        const strErrorMessage =
+            strLogPrefix + cc.fatal( "CRITICAL ERROR:" ) + " " +
+            cc.error( "BLS-raw-signer error: " ) + cc.warning( strError ) +
+            cc.error( ", stack is:" ) + "\n" + cc.stack( err.stack ) +
+            "\n";
+        log.write( strErrorMessage );
+        details.write( strErrorMessage );
+    }
+    const isSuccess = ( joSignResult && typeof joSignResult == "object" && ( !joSignResult.error ) ) ? true : false;
+    if( isExposeOutput || ( !isSuccess ) )
+        details.exposeDetailsTo( log, "BLS-raw-signer", isSuccess );
+    details.close();
+    return joSignResult;
+}
+
+export async function handle_skale_imaVerifyAndSign( joCallData ) {
```

Comment:
Please, modify function so its maximum size will be 100 codelines:
https://github.com/skalenetwork/internal-support/blob/main/docs/wiki/code-guidelines.md#function-length

Reply:
Done